### PR TITLE
add neonate data preprocessing scripts

### PR DESCRIPTION
**Pull request by suiguoxin**

### CODE REVIEW

File: docs/ neonate.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+This document describes how to process ["a dataset of neonatal EEG recordings with seizure annotations"](https:/zenodo.org/record/4940267).
+
+
+## Scripts
+
+1. Data Preprocessing
+```bash
+mkdir -p /home/gusui/Desktop/PhysioPro/data/Neonate_processed/
```
[the remark this answers is not in the document]
Comment (suiguoxin) [reply docs/ neonate.md:8]: `NeonateDataset` is not supported yet, we'll add a runnable config later.

File: scripts/montage.py [hunk 2]
```diff
@@ -25,4 +25,8 @@
 bipolar = ["FP1-F7", "F7-T3", "T3-T5", "T5-O1", "FP2-F8", "F8-T4", "T4-T6", "T6-O2", "A1-T3",
            "T3-C3", "C3-CZ", "CZ-C4", "C4-T4", "T4-A2", "FP1-F3", "F3-C3", "C3-P3", "P3-O1",
            "FP2-F4", "F4-C4", "C4-P4", "P4-O2"]
-bipolar = [s.upper() for s in bipolar]
+
+bipolar_neonate = ["Fp2-F4", "F4-C4", "C4-P4", "P4-O2", "Fp1-F3", "F3-C3", "C3-P3", "P3-O1",
+                   "Fp2-F8", "F8-T4", "T4-T6", "T6-O2", "Fp1-F7", "F7-T3", "T3-T5", "T5-O1",
+                   "Fz-Cz", "Cz-Pz"]
+bipolar_neonate = [s.upper() for s in bipolar_neonate]
```
[the remark this answers is not in the document]
Comment (suiguoxin) [reply scripts/montage.py]: > will it be better to change `bipolar` to `bipolar_tusz`? consider capitalizing `Fp1-F3` to `FP1-F3`.

Fixed, thx

File: scripts/neonate.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    return nedf, label
+
+
+def get_features_and_labels(raw_data_dir, feature_dir, label_dir, edfs, annots, duration):
+    """
+    Calculate and serialize features and labels
+    """
+    labels = {}
+    nedfs = {}
+    for f in tqdm(edfs):
```
[the remark this answers is not in the document]
Comment (suiguoxin) [reply scripts/neonate.py]: > consider supporting resumes like processing tusz dataset.

This dataset is quite small, I prefer not adding the resume part.

File: scripts/neonate.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+        meta = {'train': meta_train,
+                'test': meta_test,
+                'mean@train': mean,
+                'std@train': std,
+                'mean@test': mean_test,
+                'std@test': std_test}
+        if fft:
+            meta[mean_fft_train] = mean_fft_train
+            meta[std_fft_train] = std_fft_train
```
[the remark this answers is not in the document]
Comment (suiguoxin) [reply scripts/neonate.py]: > `meta[mean_fft_train]` should be `meta['fft_mean@train']` and similar for std, please make it corresponds to tusz processing.

Fixed, thanks

File: scripts/neonate.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+        feature_line_train_df = pd.concat(list(feature_train_set.values()), axis='index')
+        feature_line_test_df = pd.concat(list(feature_test_set.values()), axis='index')
+        label_line_train_df = pd.concat(list(label_train_set.values()), axis='index')
+        label_line_test_df = pd.concat(list(label_test_set.values()), axis='index')
+
+        merged[idx+1] = {'feature_train': feature_line_train_df,
+                         'feature_test': feature_line_test_df,
+                         'label_train': label_line_train_df,
+                         'label_test': label_line_test_df}
+        idx += 1
```
[the remark this answers is not in the document]
Comment (suiguoxin) [reply scripts/neonate.py]: > where is `merged` saved? why we have `get_mata` and `get_statistics`? why use the name statistics? can be use name like cross-validation?

Removed get_statistics. Seems not useful at all.

File: scripts/neonate.py [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        test_df = pd.concat(test_list, axis='index')
+
+        mean = train_df.mean(axis='index')
+        std = train_df.std(axis='index')
+        mean_test = test_df.mean(axis='index')
+        std_test = test_df.std(axis='index')
+
+        if fft:
+            print("Calculating FFT info...")
+            to_fft = train_df.values.reshape(-1, 256, 18).transpose(0, 2, 1)  # second_num * channels * frequency
```
[the remark this answers is not in the document]
Comment (suiguoxin) [reply scripts/neonate.py]: > `18` and `256` should not be written in the code.

Fixed, thanks

File: docs/ neonate.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (suiguoxin) [reply docs/ neonate.md:8]: Added some arguments.

File: scripts/neonate.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (chenyuqi990215) [reply scripts/neonate.py]: fine, make sense to me.

File: docs/ neonate.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chenyuqi990215) [reply docs/ neonate.md:8]: will it better if changing `/home/gusui/Desktop/PhysioPro/data/Neonate_processed/` to `data/Neonate_processed/`?

File: scripts/neonate.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (chenyuqi990215) [reply scripts/neonate.py]: why remove the functions? shall you support 4-fold cross-validation? follow the setting of [SRD](https://seqml.github.io/srd/)?

File: docs/ neonate.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (suiguoxin) [reply docs/ neonate.md]: > will it better if changing `/home/gusui/Desktop/PhysioPro/data/Neonate_processed/` to `data/Neonate_processed/`A?

Absolutely, fixed

File: scripts/neonate.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (suiguoxin) [reply scripts/neonate.py]: The cross-validation part is moved to `get_meta`

**Review by chenyuqi990215 — APPROVED** [review]
Approved

**Pull request merged**